mitodl/odl-video-service
# Use redbeat to schedule celery tasks

#### What are the relevant tickets?
Fixes #449

#### What's this PR do?
Replaces standard celerybeat with redbeat, which uses redis for scheduled tasks and does not create a `celerybeat-schedule` file.

#### How should this be manually tested?
- delete your `celerybeat-schedule` file if it exists from the root folder of the repo.
- Set `ODL_VIDEO_LOG_LEVEL=DEBUG` in your `.env` file
- `docker-compose build` (new requirement)
- `docker-compose up`
- check the celery logs.  There should be a line that looks like:
  ```
  Stored entry: <RedBeatSchedulerEntry: update-statuses cloudsync.tasks.update_video_statuses()
  ```
- Within a minute a scheduled task should run (`update_video_statuses`):
  ```
  celery_1  | [2018-04-04 21:00:00,652: INFO/MainProcess] Received task: cloudsync.tasks.update_youtube_statuses[0118781a-0dbc-475d-968b-3ab234e6e6ce]
  celery_1  | [2018-04-04 21:00:00,659: INFO/ForkPoolWorker-2] Task cloudsync.tasks.update_video_statuses[cb743063-bc00-43dc-8a5f-9b23918fa791] succeeded in 0.00625685699924361s: None
  ```
- Make sure that a new `celerybeat-schedule` file has not been created.
